packages/bsky/src/views/threads-v2.ts

```diff
@@ -57,40 +92,50 @@ const isThreadNoUnauthenticatedNode = (
 const isThreadPostNode = (node: ThreadTree): node is ThreadPostNode =>
   AppBskyUnspeccedGetPostThreadV2.isThreadItemPost(node.item.value)
 
-export type ThreadTree =
+const isThreadHiddenPostNode = (
+  node: ThreadTree,
+): node is ThreadHiddenPostNode =>
+  AppBskyUnspeccedGetPostThreadHiddenV2.isThreadHiddenItem(node.item)
```

Comment:
Rather than duck-typing `node` based on its inner contents (e.g. `node.item.$type`), what if `ThreadTree` items themselves offered a form of reflection?  I wonder if this could reduce some dependencies on implementation details such as presence of `$type` in API output.  Just an idea, not critical.

E.g. `node.isThreadHiddenPostNode` and `node.isThreadHiddenPostNode`.